pennylane/math/single_dispatch.py

```diff
@@ -476,7 +484,9 @@ def _asarray_torch(x, dtype=None, **kwargs):
 ar.register_function("torch", "diag", lambda x, k=0: _i("torch").diag(x, diagonal=k))
 ar.register_function("torch", "expand_dims", lambda x, axis: _i("torch").unsqueeze(x, dim=axis))
 ar.register_function("torch", "shape", lambda x: tuple(x.shape))
-ar.register_function("torch", "gather", lambda x, indices: x[indices])
+ar.register_function(
+    "torch", "gather", lambda x, indices, axis=0: x[:, indices] if axis == 1 else x[indices]

```

Comment:
not sure if there's a reason torch wasn't using `np.array` in this lambda before, but I'm keeping that behaviour in this PR.